README.md

```diff
@@ -14,33 +32,77 @@ Pkg.add("Interpolations")
 
 from the Julia REPL.
 
-## Usage
+## General usage
 
-*Note: the API might change in future versions, if necessary to accomodate new functionality.*
+Given an `AbstractArray` `A`, construct an "interpolation object" `itp` as
+```jl
+itp = interpolate(A, options...)
+```
+where `options...` (discussed below) controls the type of
+interpolation you want to perform.  To evaluate the interpolation at
+position `(x, y, ...)`, simply do
+```jl
+v = itp[x, y, ...]
+```
 
+Some interpolation objects support computation of the gradient, which
+can be obtained as
+```jl
+g = gradient(itp, x, y, ...)
 ```
-using Interpolations
-x = 1:5 # must be a unit range starting at 1
-coarse = sin(x)
-itp1 = interpolate(coarse, BSpline(Linear), OnGrid)
-extr1 = extrapolate(itp1, Flat)
-xeval = -pi:.01:3pi
-sinx = [extr1[x] for x in xeval]
+or, if you're evaluating the gradient repeatedly, a somewhat more
+efficient option is
+```jl
+gradient!(g, itp, x, y, ...)
 ```
+where `g` is a pre-allocated vector.
+
+`A` may have any element type that supports the operations of addition
+and multiplication.  Examples include scalars like `Float64`, `Int`,
+and `Rational`, but also multi-valued types like `RGB` color vectors.
 
-## Options
+Positions `(x, y, ...)` are n-tuples of numbers. Typically these will
+be real-valued (not necessarily integer-valued), but can also be types
```

Comment:
I'd say the positions can be *of* types - we support sending in `1.0 + 1.0du`, but not `Dual{Float64}`... It's a small  difference (in wording - bigger semantically), but I think it pays to be nit-picky in documentation :)